From bd2602f9fc4efdf158c2ee960457edf029734607 Mon Sep 17 00:00:00 2001
From: Peter Marschall <peter@adpm.de>
Date: Tue, 31 May 2011 14:16:21 +0200
Subject: [PATCH 01/26] OpenPGP: fix algorithms & key lengths calculation

Also include forgotten DO C3 in keylength calculation.
It contains the parameters for the authentication key.

Signed-off-by: Peter Marschall <peter@adpm.de>
---
 src/libopensc/card-openpgp.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/libopensc/card-openpgp.c b/src/libopensc/card-openpgp.c
index 5ba67b6..559c6c7 100644
--- a/src/libopensc/card-openpgp.c
+++ b/src/libopensc/card-openpgp.c
@@ -400,7 +400,7 @@ pgp_get_card_features(sc_card_t *card)
 		}
 
 		/* get supported algorithms & key lengths from "algorithm attributes" DOs */
-		for (i = 0x00c1; i < 0x0c3; i++) {
+		for (i = 0x00c1; i <= 0x00c3; i++) {
 			unsigned long flags;
 
 			/* Is this correct? */
-- 
1.7.5.3

